feat: articles on string types and object type inferences (#31)
LearningTypeScript/site: src/content/articles/object-properties-narrowed-before-usage.mdx

---
date: "February 17 2023"
description: "Limitations of TypeScript's type narrowing when it comes to objects."
meta: literals, narrowing, objects, primitive
---

# Why Values Aren't Narrowed Before Usage

TypeScript's type system boasts a feature called _type narrowing_.
Type narrowing means that if a block of code can only be reached when a value is a specific type, then the value will be considered that type inside the block.

> As an example, take the following `fruit` variable of type `string | undefined` (a union type indicating it can be either `string` or `undefined`.

Reviewer: [Proofreading] Hmmm...I think using "for" instead of "as a" would make the sentence clearer. 
```suggestion
For example, take the following `fruit` variable of type `string | undefined` (a union type indicating it can be either `string` or `undefined`.
```
